packages/fether-electron/src/main/index.js

```diff
@@ -63,3 +71,64 @@ app.on('quit', () => {
   pino.info('Leaving Fether');
   killParity();
 });
+
+/**
+ * Security. Intercept new-window events (i.e. `window.open`) by
+ * overriding event.newGuest without using the supplied options tag
+ * to try to mitigate risk of an exploit re-enabling node integration
+ * despite being turned off in the configuration
+ * (i.e. `nodeIntegration: false`).
+ *
+ * References:
+ * - https://www.electronjs.org/blog/webview-fix
+ * - https://blog.scottlogic.com/2016/03/09/As-It-Stands-Electron-Security.html
+ */
+app.on('web-contents-created', (event, win) => {
+  win.on(
+    'new-window',
+    (event, newURL, frameName, disposition, options, additionalFeatures) => {
+      event.newGuest = null;
+
+      if (!options.webPreferences) {
+        options.webPreferences = {};
+      }
+
+      options.webPreferences.nodeIntegration = false;
+      options.webPreferences.nodeIntegrationInWorker = false;
+      options.webPreferences.webviewTag = false;
+      delete options.webPreferences.preload;
+    }
+  );
+});
+
+/**
+ * Security. Intercept and prevent new WebView (that may be used
+ * by an attacker to gain access to the file system) in addition
+ * to setting `webviewTag: false`
+ *
+ * Reference: https://www.electronjs.org/blog/webview-fix
+ */
+app.on('web-contents-created', (eventOuter, contents) => {
+  contents.on('will-attach-webview', (eventInner, webPreferences, params) => {
+    // Strip away preload scripts if unused or verify their location is legitimate
+    delete webPreferences.preload;
+    delete webPreferences.preloadURL;
+
+    // Disable Node.js integration
+    webPreferences.nodeIntegration = false;
+
+    // Verify URL being loaded
+    if (!params.src.startsWith('http://localhost.3000/')) {
+      eventOuter.preventDefault();
+      eventInner.preventDefault();
+    }
+  });
+
+  contents.on('will-navigate', (event, navigationUrl) => {
+    const parsedUrl = parseUrl(navigationUrl);
+
+    if (parsedUrl.origin !== 'http://localhost:3000') {
```

Comment:
Weirdly enough, in the final binary, I can still navigate, while the origin is clearly not localhost:3000. I'm not sure about this one tbh.

Reply:
It looks like this is a known issue that was possibly deemed a duplicate incorrectly https://github.com/electron/electron/issues/11885